djoser/settings.py

```diff
@@ -32,35 +38,49 @@
 }
 
 
+class Settings(object):
+    def __init__(self, default_settings, explicit_overriden_settings=None):
+        if explicit_overriden_settings is None:
+            explicit_overriden_settings = {}
+
+        for setting_name, setting_value in six.iteritems(default_settings):
+            if setting_name.isupper():
+                setattr(self, setting_name, setting_value)
+
+        overriden_djoser_settings = getattr(settings, DJOSER_SETTINGS_NAMESPACE, {}) or explicit_overriden_settings
+        for overriden_setting_name, overriden_setting_value in six.iteritems(overriden_djoser_settings):
+            value = overriden_setting_value
+            if isinstance(overriden_setting_value, dict):
+                value = getattr(self, overriden_setting_name, {})
+                value.update(overriden_setting_value)
+            setattr(self, overriden_setting_name, value)
+
+
+class LazySettings(LazyObject):
+    def _setup(self, explicit_overriden_settings=None):
+        self._wrapped = Settings(default_settings, explicit_overriden_settings)
+
+
+config = LazySettings()
```

Comment:
@haxoza @piotr-szpetkowski `djoser.conf.settings` would be the best but it will introduce incompatibility for older versions. 
I'll add deprecation warnings and try to keep old interface but it should be dropped in the future.